compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -6484,9 +6484,14 @@ private STNode parseNamedOrPositionalArg() {
         STToken secondToken = peek();
         switch (secondToken.kind) {
             case EQUAL_TOKEN:
-                STNode equal = parseAssignOp();
-                STNode valExpr = parseExpression();
-                return STNodeFactory.createNamedArgumentNode(argNameOrExpr, equal, valExpr);
+                if (argNameOrExpr.kind == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                    STNode equal = parseAssignOp();
+                    STNode valExpr = parseExpression();
+                    return STNodeFactory.createNamedArgumentNode(argNameOrExpr, equal, valExpr);
+                }
+                // Treat invalid NamedArgument as a positional argument
+                argNameOrExpr = parseExpressionRhs(DEFAULT_OP_PRECEDENCE, argNameOrExpr, true, false);
+                return STNodeFactory.createPositionalArgumentNode(argNameOrExpr);
```

Comment:
L6493-L6494 is same as the default case logic. So shall we refactor like this to avoid duplication?
and then you need to extract the default case logic out of the switch.

```suggestion
                if (argNameOrExpr.kind != SyntaxKind.SIMPLE_NAME_REFERENCE) {
                    break;
                }
                STNode equal = parseAssignOp();
                STNode valExpr = parseExpression();
                return STNodeFactory.createNamedArgumentNode(argNameOrExpr, equal, valExpr);
```